src/reducers/__tests__/player.js

```diff
@@ -0,0 +1,114 @@
+import {ActionTypes, CellStates} from '../../constants';
+import reducer from '../player.js';
+
+const {INITIAL_STATE, BOARD_TILE_SELECTED, RACK_TILE_SELECTED, END_TURN} = ActionTypes;
+
+describe('src/reducers/player', function() {
+  describe('INITIAL_STATE', function() {
+    it('initializes a players rack', function() {
+      const state = reducer({}, {type: INITIAL_STATE, id: 1});
+
+      expect(state.id).toEqual(1);
+      expect(state.letters).toEqual(['a', 'b', 'c']);
+      expect(state.points).toEqual(0);
+      expect(state.selected).toEqual(-1);
+    });
+  });
+
+  describe('BOARD_TILE_SELECTED', function() {
+    it('removes the selected tile from a players rack', function() {
+      const initstate = reducer({}, {type: INITIAL_STATE, id: 1});
+      const oldstate = {...initstate, selected: 1, letters: ['a', 'b', 'c']};
+      const expectedstate = {...oldstate, selected: -1, letters: ['a', 'c']};
+      const action = {
+        type: BOARD_TILE_SELECTED,
+        player: 1,
+        turn: 1,
+        cellstate: CellStates.EMPTY
+      };
+      const state = reducer(oldstate, action);
+
+      expect(state).not.toBe(oldstate);
+      expect(state).toEqual(expectedstate);
+    });
+
+    it('does nothing if no tile is selected', function() {
+      const initstate = reducer({}, {type: INITIAL_STATE, id: 1});
+      const oldstate = {...initstate, selected: -1, letters: ['a', 'b', 'c']};
+      const action = {
+        type: BOARD_TILE_SELECTED,
+        player: 1,
+        turn: 1,
+        cellstate: CellStates.EMPTY
+      };
+      const state = reducer(oldstate, action);
+
+      expect(state).toBe(oldstate);
```

Comment:
check for deep equality as well.  the reference can be the same, but the object contents can be different.